### PR TITLE
[windows] Add windows software DSLs

### PR DESCRIPTION
**Pull request by elafarge**
This (big big big) commit updates most of our software definitions to
make them Windows compatible. In most cases, an `if` switch has been
added to use a different path, a different installation source.

The main difference is that we do not compile on Windows. Not that it
wouldn't be possible in the long term but Omnibus doesn't make it
simpler than launching shell commands for every compilation step. In
addition to that, different osftware requirements support different
compilers (Python is officialy supported only if compiled with Visual
Studio, Curl has to be compiled with MinGW...).

So what we do instead, for most cases, is just download an MSI installer
and install it in the appropriate location from the command line with
`msiexec`. For `pip` packages that don't compile on Windows (for those
who need to compile something) we use wheels from Christoph Gohlke's
awesome website: http://www.lfd.uci.edu/~gohlke/pythonlibs/ .

Finally, for `pip` packages that compile on Windows or don't require any
compilation steps, we just use the embedded `pip` to retrieve them.

NOT DONE YET BEFORE CREATING THIS PR:
- Checking that it didn't break anything on OSX and Linux

### CODE REVIEW

File: config/software/adodbapi.rb [hunk 1]
```diff
@@ -7,5 +7,9 @@
 
 build do
   ship_license "LGPLv2"
-  command "#{install_dir}/embedded/bin/pip install -I --install-option=\"--install-scripts=#{install_dir}/bin\" #{name}==#{version}"
-end
\ No newline at end of file
+  if ohai['platform'] == 'windows'
+    command "#{install_dir}/embedded/Scripts/pip install -I --install-option=\"--install-scripts=#{install_dir}/bin\" #{name}==#{version}"
```
Comment (remh) [inline config/software/adodbapi.rb]: can't we use a new variable `pip_location` that we could use so we don't have this if statement for all python dependencies?

**Comment by olivielpeau** [comment]
@degemer Is this PR still relevant or are you working on another one?

**Comment by degemer** [comment]
@olivielpeau I'm working on another branch (https://github.com/DataDog/omnibus-software/commits/quentin/omnibus-5.0), no PR open yet. This PR is not relevant for omnibus 5, but please do not delete the 'etienne/windows' branch.

**Comment by remh** [comment]
Closing this PR then.

**Pull request closed without merging**